DDS-901_consistency_db_data_migrate

dmlond opened the pull request.

dmlond committed:
Merge tag 'v1.3.21' into DDS-901_consistency_db_data_migrate

Bug fixes for /app/status

dmlond committed:
added data migration to update is_consistent status for project and upload
appropriately.

dmann commented on spec/lib/tasks/db_data_migrate_rake_spec.rb:
```diff
@@ -141,5 +141,66 @@
+
+      context 'record is not consistent' do
+        it 'should update is_consistent to false' do
+          invoke_task
```
Is there a reason that you're not checking `record_class.where(is_consistent: nil)` like you're doing in the above it block?

dmlond replied:
no :)

dmann replied:
I meant on both sides of the `invoke_task` call...  :P

dmlond replied:
ok. should be good to go

dmann commented on spec/lib/tasks/db_data_migrate_rake_spec.rb:
```diff
@@ -141,5 +141,66 @@
+      context 'record is consistent' do
+        let(:prep_method) { send(prep_method_sym) }
+        before do
+          prep_method.call
```
Using a `Proc` and having to use `.call` doesn't really fit any of our existing patterns. I believe you could get the same result without a `Proc` and by replacing the above with: `expect { proc_method }.not_to raise_error`

Which gives the added bonus of trapping possible exceptions when `prep_method` is executed.

dmann commented on spec/lib/tasks/db_data_migrate_rake_spec.rb:
```diff
@@ -141,5 +141,66 @@
+        let(:init_project_storage) {
+          Proc.new do
+            storage_provider.put_container(record.id)
+            expect(storage_provider.get_container_meta(record.id)).to be
```
I've been trying to get away from using `.to be` without an argument. It's kinda vague and has lead to confusion when going over the tests with new developers. If you're trying to assert that something is returned, use `.not_to be_nil` instead.

dmann commented on spec/lib/tasks/db_data_migrate_rake_spec.rb:
```diff
@@ -141,5 +141,66 @@
+        let(:init_upload) {
+          Proc.new do
+            storage_provider.put_container(record.project.id)
+            expect(storage_provider.get_container_meta(record.project.id)).to be
```
See above regarding `.to be`

dmann commented on spec/lib/tasks/db_data_migrate_rake_spec.rb:
```diff
@@ -141,5 +141,66 @@
+            storage_provider.put_container(record.project.id)
+            expect(storage_provider.get_container_meta(record.project.id)).to be
+            record.create_and_validate_storage_manifest
+            record.update(is_consistent: nil)
```
Be aware that using `update` will run validations. If you want to skip validations, use `update_attribute`. If you also want to skip callbacks, use `update_column`. If you don't care about either, use `update!` so that an exception is thrown if something doesn't work as expected.

dmann commented on spec/lib/tasks/db_data_migrate_rake_spec.rb:
```diff
@@ -141,5 +141,66 @@
+            expect(storage_provider.get_container_meta(record.project.id)).to be
+            record.create_and_validate_storage_manifest
+            record.update(is_consistent: nil)
+            expect(storage_provider.get_object_metadata(record.project.id, record.id)).to be
```
See above regarding `.to be`

dmann reviewed (changes requested):
Looks good for the most part, just a few changes.

dmlond committed:
simpler use of let without the extra procs

dmlond committed:
addresses the rest of the code review change requests
 - to be => not_to be_nil
 - uses update_columns instead of update to avoid unecessary validations and callbacks

dmlond commented:
ready for final review

dmlond committed:
last code review change request addressed
